### PR TITLE
fix: fix non terminating PositionBuilder

### PR DESCRIPTION
**Pull request by Strum355**
PositionBuilder would get caught in the `<` in the annotation string, resulting in `o2` == `o1` and no advancement being made.
This fix ignores the opening angle bracket if it comes at the start of the search space

### CODE REVIEW

File: src/test/java/spoon/test/position/PositionTest.java [hunk 1]
```diff
@@ -424,6 +426,13 @@ public void testPositionGeneric() throws Exception {
 		assertEquals("protected static", contentAtPosition(classContent, position3.getModifierSourceStart(), position3.getModifierSourceEnd()));
 	}
 
+	@Test(timeout=10000)
```
Comment (MartinWitt) [inline src/test/java/spoon/test/position/PositionTest.java:429]: is there a reason for the timeout?

Comment (Strum355) [reply src/test/java/spoon/test/position/PositionTest.java:429]: The test specifically tests to make sure PositionBuilder terminates in a reasonable time. The existing bug caused non-termination, therefore if it doesn't terminate in a reasonable time, we assume it to be non-terminating

**Comment by monperrus** [comment]
LGTM, all tests are green, will merge.

**Comment by monperrus** [comment]
Thanks a lot @Strum355

**Pull request merged**